File: scalaleetcode/src/qizy/algorthim/BubbleSort.scala
package qizy.algorthim

object BubbleSort {

  def bubblesort(a: Array[Int]) = {
    for(i<- 0 until a.length-1){
      for(j<- 0 until a.length-1-i){
        if(a(j)<=a(j+1)){
          val t =a(j)
          a(j)=a(j+1)
          a(j+1)=t
        }
      }
    }
  }


  def main(args:Array[String]) {
    val a = Array(4,1,6,2,8,9,11,3,5)
    bubblesort(a)
    for( i<- 0 until a.length){
      println(a(i))
    }
  }
}

File: javaleetcode/src/main/java/qizy/algorithm/sort/BubbleSort.java
package qizy.algorithm.sort;

public class BubbleSort {

	public static void main(String[] args) {
		int[] a = {5,3,4,8,2,7,1,9,11,6,5};
		bubblesort(a);
		SortUtil.print(a);
	}

	private static void bubblesort(int[] a) {
		for(int i=0; i<a.length-1;i++) {
			for(int j=0;j<a.length-1-i;j++) {
				if(a[j]>=a[j+1]) {
					SortUtil.swap(j,j+1,a);
				}
			}
		}
		
	}

	

	
}
